pkg/cloud/aws/actuators/machine/actuator_test.go

```diff
@@ -292,3 +293,57 @@ func mockTerminateInstances(mockAWSClient *mockaws.MockClient) {
 	mockAWSClient.EXPECT().TerminateInstances(gomock.Any()).Return(
 		&ec2.TerminateInstancesOutput{}, nil)
 }
+
+func TestRemoveDuplicatedTags(t *testing.T) {
+	cases := []struct {
+		tagList  []*ec2.Tag
+		expected []*ec2.Tag
+	}{
+		{
+			// empty tags
+			tagList:  []*ec2.Tag{},
+			expected: []*ec2.Tag{},
+		},
+		{
+			// no duplicate tags
+			tagList: []*ec2.Tag{
+				{Key: aws.String("clusterID"), Value: aws.String("test-ClusterIDValue")},
+				{Key: aws.String("clusterName"), Value: aws.String("test-ClusterNameValue")},
+			},
+			expected: []*ec2.Tag{
+				{Key: aws.String("clusterID"), Value: aws.String("test-ClusterIDValue")},
+				{Key: aws.String("clusterName"), Value: aws.String("test-ClusterNameValue")},
+			},
+		},
+		{
+			// multiple duplicate tags
+			tagList: []*ec2.Tag{
+				{Key: aws.String("clusterID"), Value: aws.String("test-ClusterIDValue")},
+				{Key: aws.String("clusterName"), Value: aws.String("test-ClusterNameValue")},
+				{Key: aws.String("clusterSize"), Value: aws.String("test-ClusterSizeValue")},
+				{Key: aws.String("clusterName"), Value: aws.String("test-ClusterNameDuplicatedValue")},
+				{Key: aws.String("clusterSize"), Value: aws.String("test-ClusterSizeDuplicatedValue")},
+			},
+			expected: []*ec2.Tag{
+				{Key: aws.String("clusterID"), Value: aws.String("test-ClusterIDValue")},
+				{Key: aws.String("clusterName"), Value: aws.String("test-ClusterNameValue")},
+				{Key: aws.String("clusterSize"), Value: aws.String("test-ClusterSizeValue")},
+			},
+		},
+	}
+
+	for i, c := range cases {
+		actual := removeDuplicatedTags(c.tagList)
+		if !reflect.DeepEqual(c.expected, actual) {
+			t.Errorf("test case %d: output of removeDuplicatedTags is not as expected.\n", i)
```

Comment:
I think it would be more succinct to reduce all of this output to just:

```
t.Errorf("test #%d: expected %+v, got %+v", c.expected, actual)
```

as the inputs are quite small.